llvm/test/CodeGen/Hexagon/hexagon-copy-hoisting.mir

```diff
@@ -11,23 +11,19 @@
 
 ---
 name:            f0
-tracksRegLiveness: true
+tracksRegLiveness: false
```

Comment:
The HexagonCopyHoisting pass does not deal with register liveness, so disabling it for this test doesn't affect its behavior. I suspect that many of the other Hexagon MIR tests have it enabled for the same reason I did, which is that it is on by default when using llc with -stop-before.

Reply:
So if it's on when you run -stop-before, maybe the pass should deal with liveness?

The pass even does
```
  // Re-compute liveness
  if (Changed) {
    LiveIntervals &LIS = getAnalysis<LiveIntervals>();
    SlotIndexes *SI = LIS.getSlotIndexes();
    SI->releaseMemory();
    SI->runOnMachineFunction(Fn);
    LIS.releaseMemory();
    LIS.runOnMachineFunction(Fn);
  }
```
Yet, liveness is broken after the pass?

Why do you (try to) recompute liveness if liveness shouldn't be tracked? And why is liveness then still broken after the pass. Something doesn't look right here.

Reply:
I maybe misspoke a bit when I said that the pass doesn't deal with liveness. Recomputing the liveness is important for other passes further down the line - if a register is only live going into a basic block because of the copy instruction which has now been moved to its predecessor, it should not be marked as live in that block. However, a register being live or not does not affect the optimization made in the HexagonCopyHoisting pass itself.

The test, on the other hand, has no need to deal with liveness at all, as we are only running the HexagonCopyHoisting pass, which itself does not use this information.

Reply:
```
Recomputing the liveness is important for other passes further down the line
```
Yeah that's why I thought it's no good if liveness is still broken after the pass? Which the verifier said?